Symfony/CS/Tests/FileCacheManagerIntegrationTest.php

```diff
@@ -0,0 +1,108 @@
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Tests;
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\SplFileInfo as FinderSplFileInfo;
+use Symfony\CS\FileCacheManager;
+use Symfony\CS\Fixer;
+
+class FileCacheManagerIntegrationTest extends \PHPUnit_Framework_TestCase
+{
+    protected function setUp()
+    {
+        $this->fs = new Filesystem();
+        $this->rootDir = __DIR__.DIRECTORY_SEPARATOR.'Fixtures'.DIRECTORY_SEPARATOR.'FileCacheManagerIntegrationTest';
+
+        $this->fs->mkdir($this->rootDir);
+        $this->isCacheAvailable = $this->isCacheAvailable();
+
+        $this->clearFileCacheManagerIntegrationTestFiles();
+
+        $this->customConfigDir = $this->rootDir.DIRECTORY_SEPARATOR.'custom_config_dir';
+        $this->fileOne = $this->rootDir.DIRECTORY_SEPARATOR.'path_one'.DIRECTORY_SEPARATOR.'CommonFilename.php';
+        $this->fileTwo = $this->rootDir.DIRECTORY_SEPARATOR.'path_two'.DIRECTORY_SEPARATOR.'CommonFilename.php';
+
+        $this->fs->mkdir($this->rootDir);
+        $this->fs->mkdir($this->customConfigDir);
+        $this->fs->mkdir(dirname($this->fileOne));
+        $this->fs->mkdir(dirname($this->fileTwo));
+
+        file_put_contents($this->fileOne, "<?\n// one\n");
+        file_put_contents($this->fileTwo, "<?\n// two\n");
+    }
+
+    protected function tearDown()
+    {
+        $this->clearFileCacheManagerIntegrationTestFiles();
+    }
+
+    private function clearFileCacheManagerIntegrationTestFiles()
+    {
+        $this->fs->remove($this->rootDir);
+    }
+
+    private function isCacheAvailable()
+    {
+        $fileCacheManager = new FileCacheManager(true, $this->rootDir, array());
+        $reflectionIsCacheAvailable = new \ReflectionMethod($fileCacheManager, 'isCacheAvailable');
+        $reflectionIsCacheAvailable->setAccessible(true);
+        $isCacheAvailable = $reflectionIsCacheAvailable->invoke($fileCacheManager);
+        $reflectionIsCacheAvailable->setAccessible(false);
+        unset($reflectionIsCacheAvailable, $fileCacheManager);
+
+        return $isCacheAvailable;
+    }
+
+    /**
+     * @dataProvider provideScenarios
+     */
+    public function testCacheScenarios($insideCacheFileDir)
+    {
+        if (!$this->isCacheAvailable) {
+            $this->markTestSkipped('Cache functionality not supported');
+        }
+
+        $configDir = $this->rootDir;
+        if (!$insideCacheFileDir) {
+            $configDir = $this->customConfigDir;
+        }
+
+        $fixer = new Fixer();
+        $fixer->addFixer(new \Symfony\CS\Fixer\PSR1\ShortTagFixer());
+        $fixers = $fixer->getFixers();
+
+        $splFileOne = new FinderSplFileInfo($this->fileOne, dirname($this->fileOne), basename($this->fileOne));
+        $splFileTwo = new FinderSplFileInfo($this->fileTwo, dirname($this->fileTwo), basename($this->fileTwo));
```

Comment:
Following the comment above, how about just mapping the field `$fieldNames` like this:

```php
/* @var FinderSplFileInfo[] $files */
$files = array_map(function ($fileName) {
    return new FinderSplFileInfo(
        $fileName,
        dirname($fileName),
        basename($fileName)
    );
}, $this->fileNames);
```

Then we could just iterate over the `$files` array below, both for warming up the cache, and asserting that the files do not need fixing. What do you think?